include/xtensor/xmath.hpp

```diff
@@ -2143,7 +2143,7 @@ namespace detail {
     }
 
     template <class E, class X, class EVS = DEFAULT_STRATEGY_REDUCERS,
-              XTL_REQUIRES(xtl::negation<is_reducer_options<X>>, xtl::negation<std::is_integral<std::decay_t<X>>>, xtl::negation<std::is_integral<EVS>>)>
+              XTL_REQUIRES(xtl::negation<is_reducer_options<X>>, xtl::negation<std::is_integral<std::decay_t<X>>>)>
```

Comment:
I guess we should require `EVS` to be `reducer_options` ? otherwise we may run in the same problem as the one this PR fixes for axes passed as an array

Reply:
The test code

```
xt::variance(py_a, {0, 1, 2, 3}, 1);
```

In the old version the sequence of calls is

```
inline auto variance(E&& e, const A (&axes)[N], EVS es = EVS()) ->
inline auto variance(E&& e, X&& axes, D const& ddof, EVS es = EVS())
```
because `xtl::negation<std::is_integral<EVS>>` prevents  
```
inline auto variance(E&& e, X&& axes, EVS es = EVS())
```
, which is a natural successor, from being called. So it chose the wrong path at be beginning (take 1 as EVS) but in the end was navigated to the correct one (take 1 as ddof). Conversely, the wrong path was chosen because one overload is missing.

The missing overload is added in the new version, thus the sequence of calls is
```
inline auto variance(E&& e, const A (&axes)[N], D const& ddof, EVS es = EVS()) -> 
inline auto variance(E&& e, X&& axes, D const& ddof, EVS es = EVS())
````
Therefore, `xtl::negation<std::is_integral<EVS>>` is no more needed and it makes the template looks bloated. And I cannot think of a case which would need it.

Still, the downside of the current implementation is that it does not support axes being an integer. So in

```
xt::variance(py_a, 1);
```
1 is interpreted as `ddof` and 

```
xt::variance(py_a, 1, 1);
```
will not compile.

Reply:
I was thinking of an issue with this kind of call:

```cpp
std::vector<size_t> x = {0, 1, 2, 3};
xt::variance(py_a, x, 1);
```

which becomes ambiguous.

Reply:
You are absolutely right! Thanks! I also added the unittest to cover this.